helix-view/src/document.rs

```diff
@@ -209,6 +310,189 @@ impl Document {
         }
     }
 
+    pub fn set_encoding(&mut self, label: &str, view_id: ViewId) -> Result<(), anyhow::Error> {
+        match encoding_rs::Encoding::for_label(label.as_bytes()) {
+            Some(encoding) => self._set_encoding(encoding, view_id)?,
+            None => return Err(anyhow::anyhow!("unknown encoding")),
+        };
+
+        Ok(())
+    }
+
+    fn _set_encoding(
```

Comment:
Yeah, I'm actually having a hard time wrapping my head around this code, actually.  (Sorry kirawi! :-/)

From the name, I would expect `_set_encoding()` (and `set_encoding()`) to just change the encoding setting in the document.

And that leaves me not entirely sure what `_set_encoding()` is actually for (possibly because I haven't been following the conversation closely enough).  Is it reloading the document from disk with a different encoding?  If so, then `reload()` or something like that would definitely be more descriptive.

I also just generally agree with the workflow @archseer outlined above.  That's how I would expect it to work as a user.